### PR TITLE
Setting page cleanup

### PR DESCRIPTION
**Pull request by gguijarro-c-chwy**
### 🕵️‍♂️ Notes for Code Reviewer

Settings page cleanup.

### 🙈 Screenshots

image

### 👯‍♀️ Paired with

Solo

### CODE REVIEW

File: lib/screens/settings/settings_screen.dart [hunk 1]
```diff
@@ -26,58 +26,36 @@ class SettingsScreen extends StatelessWidget {
       body: BlocProvider(
         create: (context) =>
             SettingsBloc(BlocProvider.of<AuthenticationBloc>(context))..add(const SetUpInitialValues()),
-        child: MultiBlocListener(
```
Comment (gguijarro-c-chwy) [inline lib/screens/settings/settings_screen.dart:29]: we should never have `MultiBlocListener` to listen to the same bloc 'AuthenticationBloc'

You only use `MultiBlocListener` when you need to listen to the state in different blocs.

File: lib/screens/settings/settings_screen.dart [hunk 2]
```diff
@@ -26,58 +26,36 @@ class SettingsScreen extends StatelessWidget {
...
-                NavigationService.of(context).navigateTo(Routes.verification).then((isValid) {
-                  if (isValid ?? false) {
-                    BlocProvider.of<SettingsBloc>(context).add(const OnValidVerification());
-                  }
-                });
-              },
-            ),
-            BlocListener<SettingsBloc, SettingsState>(
-              listenWhen: (previous, current) =>
-                  previous.isSecureBiometric == false && current.isSecureBiometric == true,
```
Comment (gguijarro-c-chwy) [inline lib/screens/settings/settings_screen.dart:52]: previous.isSecureBiometric == false && current.isSecureBiometric == true,

All these checks should not be here, they should all happen in the bloc, then send page commands for the UI to react.

File: lib/screens/settings/interactor/viewmodels/settings_state.dart [hunk 3]
```diff
@@ -9,28 +9,22 @@ class SettingsState extends Equatable {
   final PageCommand? pageCommand;
   final String? errorMessage;
   final bool hasNotification;
-  final bool? navigateToGuardians;
```
Comment (gguijarro-c-chwy) [inline lib/screens/settings/interactor/viewmodels/settings_state.dart:12]: Actions are not State, they should be represented by Page Commands. 

State - It holds the UI state, What should the UI draw. 

Navigate to X is the perfect example of an action and not a state. The UI doesnt need to draw anything, it needs to navigate. 

Show dialog is another example of the Action and not a state.

File: lib/screens/settings/settings_screen.dart [hunk 4]
```diff
@@ -26,58 +26,36 @@ class SettingsScreen extends StatelessWidget {
...
-                          child: const LogoutRecoveryPhraseDialog(),
-                        );
-                      },
-                    ).whenComplete(() => BlocProvider.of<SettingsBloc>(context).add(const ResetShowLogoutButton()));
-                  } else if (pageCommand is NavigateToRoute) {
-                    NavigationService.of(context).navigateTo(pageCommand.route);
-                  }
-                }),
-          ],
+        child: BlocListener<SettingsBloc, SettingsState>(
```
Comment (gguijarro-c-chwy) [inline lib/screens/settings/settings_screen.dart:29]: All the above code was removed in favor or `BlocListener` and Page Commands  over items in the state.

**Review by n13 — APPROVED** [review]
Much appreciated, thank you

**Comment by n13** [comment]
Something is wrong with the GH linter action, it never finishes, I will just merge this

**Pull request merged**